### DISCUSSION TITLE
adding new epochs

### DISCUSSION THREAD

**Discussion by ghost** [ultralytics/ultralytics#10110] [Q&A]
Hi, I trained a yolov8 model. Now, i want to add new epochs and i want to continue that training. My aim was increasing epoch size. (I tried resume= True but i had an error 'epochs is finished, nothing to resume. start new training without resuming'. When i loaded last.pt and started new training, it was looking like whole train started new. ) Even i tried to change args.yaml file.

**Comment by glenn-jocher** [comment]
Hello! 😊

It looks like you're trying to extend your training beyond its original epoch limit. To achieve this without starting from scratch, you need to manually adjust the `epochs` value in your training script and use the checkpoint file (`last.pt`) to continue training. Here's a quick guide:

1. Make sure `epochs` in your training command is set to the new total number of epochs you desire.
2. Use the `last.pt` file from your previous training as the `model` parameter.
3. Ensure you're not setting `resume=True` since you're changing the total epochs.

Here's an example command:

```bash
yolo train data=coco128.yaml model=path/to/last.pt epochs=NEW_EPOCH_TOTAL imgsz=640
```

Just replace `NEW_EPOCH_TOTAL` with your new target number of epochs and provide the correct path to your `last.pt` from the initial training phase.

It's essential not to modify the `args.yaml` directly for this purpose as it won't affect the epochs parameter passed during training.

Hope this helps you to seamlessly continue your training! If you encounter any further issues, feel free to ask.

**Reply by ghost** [reply]
Hi,
I just started the train. I want to consult two more things. 

First of all, if i will use checkpoint epoch900.pt instead of last.pt, will it impact training?
Then, for first train i decided lr0=0.01, but now should i change it?

Thank you.

**Reply by glenn-jocher** [reply]
@elifyanar hello! 😊

Starting with your first question, yes, choosing `epoch900.pt` as a checkpoint will impact training since it resumes from that specific epoch's state. Your model will continue learning from epoch 901 onwards, preserving its learned weights and optimization state from that point.

Regarding your second question, it's not strictly necessary to change `lr0=0.01` when resuming training, unless your initial results suggested that a different learning rate might be more effective. If you've observed that your model was learning well with that setting, it might be just fine to keep it as is.

Feel free to experiment, and happy training!

```bash
# Example command to resume training from epoch900.pt with the same learning rate:
yolo train data=coco128.yaml model=path/to/epoch900.pt epochs=TOTAL_EPOCHS imgsz=640 lr0=0.01
```

Best of luck with your training! 🚀

**Reply by ghost** [reply]
![Capture](https://github.com/ultralytics/ultralytics/assets/83651273/b56c9096-c554-4f40-896b-49044d292d55)
Hello,

I shared a photo. I followed your steps and it is result. But i think something is going wrong. Why new train is not using old weights?

**Reply by glenn-jocher** [reply]
@elifyanar hello! 😊 It seems like there might have been an issue with how the training process was resumed. To ensure the new training utilizes the old weights correctly, you should explicitly specify the checkpoint file (e.g., `last.pt` or `epoch900.pt`) using the `model` argument in your command. Here's a quick example:

```bash
yolo train data=coco128.yaml model=path/to/your_checkpoint.pt epochs=TOTAL_EPOCHS imgsz=640
```

Make sure the path to your checkpoint file is correct. This should have the training resume using the weights you've previously trained. If problems persist, double-check the checkpoint file's integrity and path. Happy training!

**Reply by ghost** [reply]
Hi, 
I followed all steps but solution is same for my case.  I am trying to sort it out, so i will ask a question, i am using my custom dataset. Can it impact? I believe that there is a solution for my case because i just want to extend epoch size. Plese, can you give me support?
Thank you.

**Reply by glenn-jocher** [reply]
@elifyanar hello! 😊 Using a custom dataset should not hinder the process of extending your epoch size. To adjust the epochs and continue training from a checkpoint, ensure you specify both the `model` with your checkpoint file and the new `epochs` value in your command. For instance:

```bash
yolo train data=your_custom_dataset.yaml model=path/to/checkpoint.pt epochs=new_epoch_count imgsz=640
```

Just replace `your_custom_dataset.yaml`, `path/to/checkpoint.pt`, and `new_epoch_count` with your actual dataset config, checkpoint path, and desired total epochs. This should help you seamlessly continue training with the desired epoch extension on your custom dataset. Let's tackle this together!
